cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java

```diff
@@ -52,11 +67,47 @@ public HttpExecutorImpl(HttpClient client) {
         this.httpClient = client;
     }
 
+    public Organization getOrganization(URI uri) throws ApiGatewayException {
+        return fromSubSubunit(getSubSubUnitDtoWithMultipleEfforts(uri));
+    }
+
+    protected SearchResponse<Organization> query(URI uri) throws NotFoundException, FailedHttpRequestException {
+        HttpResponse<String> response = sendRequestMultipleTimes(addLanguage(uri)).get();
+        if (isSuccessful(response.statusCode())) {
+            try {
+                List<Organization> organizations = getOrganizations(response);
+                return new SearchResponse<Organization>(uri)
+                        .withHits(organizations)
+                        .withSize(getCount(response, organizations));
+            } catch (JsonProcessingException e) {
+                throw new FailedHttpRequestException(e.getMessage());
+            }
+        } else if (response.statusCode() == HTTP_NOT_FOUND) {
+            throw new NotFoundException(String.format(NOT_FOUND_MESSAGE_TEMPLATE, uri));
+        } else {
+            throw new FailedHttpRequestException(errorMessage(response));
+        }
+    }
+
+
+    private Organization getParentOrganization(SubSubUnitDto subSubUnitDto) {
+        URI parent = Optional.ofNullable(subSubUnitDto.getParentUnit()).map(InstitutionDto::getUri).orElse(null);
+        final Set<Organization> partOf = isNull(parent)
+                ? null
+                : Set.of(getParentOrganization(attempt(() ->
```

Comment:
Can this go into an infinite loop?

Reply:
Only if there is a loop in the data in cristin

Reply:
Good thing we have a timeout in lambda

Reply:
I runs out of memory, have tried that....